### PR TITLE
CB2-12965: Refactor test result and technical record logs

### PR DESCRIPTION
**Pull request by Daniel-Searle**
## Refactor test result and technical record logs

In this change, I have refactored the way that the logging system is working and fixed issues with technical record statusCode not appearing as well as adding additional unit tests.

Related issue: [CB2-12965](https://dvsa.atlassian.net/browse/CB2-12965)

## Before submitting (or marking as "ready for review")

- [ ] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: tests/unit/functions/process-stream-event.unitTest.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ jest.mock('../../../src/services/entity-conversion', () => ({
 describe('processStreamEvent()', () => {
   beforeEach(() => {
     jest.resetAllMocks();
+    jest.restoreAllMocks()
```
Comment (cb-cs) [inline tests/unit/functions/process-stream-event.unitTest.ts]: missing ;

File: src/functions/process-stream-event.ts [hunk 2]
```diff
@@ -101,11 +104,12 @@ export const processStreamEvent: Handler = async (
 
         await convert(tableName, operationType, image);
```
Comment (cb-cs) [inline src/functions/process-stream-event.ts:105]: What happens if an error is thrown here and caught in the catch block, prior to printLogs() being called? Is there any information in the ILog that could still be useful, or are we happy there is enough information in the error logs in the catch block?

Comment (Daniel-Searle) [reply src/functions/process-stream-event.ts:105]: The SQS message id gets logged on error and the debug logs can be switched on if there is an issue that needs investigating. Happy to get the general consensus on it though! 😸

File: src/functions/process-stream-event.ts [hunk 3]
```diff
@@ -11,12 +11,12 @@ import { BatchItemFailuresResponse } from '../models/batch-item-failure-response
 import { destroyConnectionPool } from '../services/connection-pool';
 import { DynamoDbImage } from '../services/dynamodb-images';
 import { convert } from '../services/entity-conversion';
-import { debugLog } from '../services/logger';
+import {addToLogManager, clearLogs, createLogEntry, debugLog, printLogs, updateLogEntry} from '../services/logger';
```
Comment (cb-cs) [inline src/functions/process-stream-event.ts]: Can you format these imports please

**Review by cb-cs — APPROVED** [review]

**Review by m-mullen — APPROVED** [review]

**Pull request merged**